core/FrontController.php

```diff
@@ -418,6 +419,10 @@ public function init()
             $this->makeAuthenticator($sessionAuth); // Piwik\Auth must be set to the correct Login plugin
         }
 
+        if ($this->isSupportedBrowserCheckNeeded()) {
```

Comment:
@flamisz sorry I just realise we will need to move this somewhere else in the frontcontroller. Maybe in the `dispatch` method or maybe we only execute the check when `defined('PIWIK_ENABLE_DISPATCH') && PIWIK_ENABLE_DISPATCH === true` which I think would work too.

Problem: In our Matomo for WordPress plugin we call the `init()` method whenever we need to bootstrap Matomo "headless" see https://github.com/matomo-org/wp-matomo/blob/4.1.3/classes/WpMatomo/Bootstrap.php#L41-L78  In that case we disable the dispatch and session start see https://github.com/matomo-org/wp-matomo/blob/4.1.3/classes/WpMatomo/Bootstrap.php#L49-L56

If we were to leave the check, and Matomo is bootstrapped as part of a regular website page load of a regular Matomo user, then we'd break their websites on some requests for older browsers.

I'm pretty sure the other WP-Matomo plugin does the same and I'm pretty sure we recommended for quite a while (or we still do somewhere maybe) calling the frontcontroller init method to bootstrap Matomo "headless" and would therefore also break other parts. You can see how this used to be done quite often (and probably still many do) eg here: https://forum.matomo.org/t/show-the-piwik-data-in-a-php-page/20230/2

Not sure if it's better to move it into the dispatch method or checking for the constant. It should be fine to simply not execute the check if `PIWIK_ENABLE_DISPATCH===false` I think.

Reply:
ok, I check this and modify